demoproviders/barbican/java/org/openyolo/demoprovider/barbican/provider/DeleteCredentialActivity.java

```diff
@@ -0,0 +1,168 @@
+            return;
+        }
+
+        mCallerAuthDomains = CallerUtil.extractCallerAuthDomains(this);
+
+        if (mCallerAuthDomains.isEmpty()) {
+            Log.w(LOG_TAG, "Unable to identify calling app");
+            setResultAndFinish(CredentialDeleteResult.BAD_REQUEST);
+            return;
+        }
+
+        // TODO: expand the caller auth domains to all equivalent domains
+
+        if (!mCallerAuthDomains.contains(mRequest.getCredential().getAuthenticationDomain())) {
+            setResultAndFinish(CredentialDeleteResult.BAD_REQUEST);
+            return;
+        }
+
+        CredentialStorageClient.connect(this, this);
+    }
+
+    @Override
+    public void onStorageConnected(CredentialStorageClient client) {
+        mStorageClient = client;
+
+        final Credential credentialToDelete = mRequest.getCredential();
+
+        if (!mStorageClient.isCreated()) {
+            Log.i(LOG_TAG, "Store is not yet initialized");
+            setResultAndFinish(CredentialDeleteResult.NO_MATCHING_CREDENTIAL);
+            return;
+        }
+
+        try {
+            Protobufs.Credential proto = credentialToDelete.toProtobuf();
+            if (!mStorageClient.hasCredential(proto)) {
+                setResultAndFinish(CredentialDeleteResult.NO_MATCHING_CREDENTIAL);
+                return;
+            }
+        } catch (IOException ex) {
+            setResultAndFinish(CredentialDeleteResult.UNKNOWN);
+        }
+
+        setContentView(R.layout.delete_layout);
+        ButterKnife.bind(this);
+
+        // monitor interactions outside of the dialog
+        getWindow().setFlags(
+                LayoutParams.FLAG_NOT_TOUCH_MODAL,
+                LayoutParams.FLAG_NOT_TOUCH_MODAL);
+        getWindow().setFlags(
+                        LayoutParams.FLAG_WATCH_OUTSIDE_TOUCH,
+                        LayoutParams.FLAG_WATCH_OUTSIDE_TOUCH);
+
+        String callingAppName = CallerUtil.getCallingAppName(this);
+        String savePromptTemplate = getString(R.string.delete_prompt_template);
+        mDeletePromptView.setText(String.format(
+                savePromptTemplate,
+                credentialToDelete.getIdentifier(),
+                callingAppName).trim());
+        mAcceptDeleteButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    mStorageClient.deleteCredential(credentialToDelete.toProtobuf());
+                    setResultAndFinish(CredentialDeleteResult.DELETED);
+                } catch (IOException e) {
+                    setResultAndFinish(CredentialDeleteResult.UNKNOWN);
+                }
+            }
+        });
+
+        mRefuseDeleteButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                setResultAndFinish(CredentialDeleteResult.USER_REFUSED);
```

Comment:
Nit: Add //TODO: Mark the authentication domain/credential combination as user refused so future requests may be  automatically rejected